pkg/snapshotter/snapshotter_csi.go

```diff
@@ -1052,3 +1083,174 @@ func (c *csiDriver) CleanUpRestoredResources(backupLocation *storkapi.BackupLoca
 
 	return nil
 }
+
+func (c *csiDriver) createJob(pvc *v1.PersistentVolumeClaim, namespace string) (*batchv1.Job, error) {
+	jobName := toBoundJobPVCName(pvc.Name, string(pvc.GetUID()))
+	// if already the job is running or in completed state , no need to rerun the job
+	job, err := batch.Instance().GetJob(jobName, namespace)
+	if err == nil {
```

Comment:
If there is an error we still proceed with another part of the code which might be a problem.
Need to handle error something like this 

`if err != nil && !apierrors.IsNotFound(err)`

Also, we can't just rely on `err == nil` to know a job is completed 
Let's take a case where:
1. First-time `CreateJob()` is called, API gives no error but eventually job fails for some reason 
2. Next time the reconciler comes up and again enters `createJob()` in which case `GetJob()` doesn't give an error and if we return from here then we would never start the job again.

Jobs status needs to be fetched as follows
```
var jobStatus batchv1.JobConditionType
	if len(job.Status.Conditions) != 0 {
		jobStatus = job.Status.Conditions[0].Type

	}
```
You can refer to `JobStatus()` code in KDMP.

Reply:
No, we are not going to check for the job status here.
As mentioned earlier , the job to fail the pod needs to get bound and thats essentially we want.

But also ideally createjob should have proceeded if there is error and the error should be also NotFound type.
Addressed in the next commit where if the error is of NotFound kind , createJob will return error.